### ISSUE TITLE
bug: running `MCore`with `--iter 0` does not skip refinement

### ISSUE THREAD

**Issue by alisterburt** [warpem/warp#111]
Running `MCore`with `--iter 0` and some refinement ran and I got to higher res than I was expecting 😆 

```sh
MCore \
--population m/10491.population \
--perdevice_refine 4 \
--iter 0
```

```txt
2024-04-01 19:53:43.756 Received "MPARefine", with 4 arguments, for GPU #0, 75983 MB free:
2024-04-01 19:53:43.903 Loading tilt series and masks...
2024-04-01 19:54:07.952 61081 MB after loading raw data of TS_17.tomostar
2024-04-01 19:54:07.969 Calculating spectral weights...
2024-04-01 19:54:09.584 60701 MB after spectra estimation of TS_17.tomostar
2024-04-01 19:54:09.596 Precomputing gradient weights...
2024-04-01 19:54:11.313 Re-extracting particles for optimization iteration 1/3
2024-04-01 19:54:11.832 Running optimization iteration 1/3, image & volume warping34792.79271988038
34835.02387077313
34805.559197038645
34841.12636432571
34829.692000728835
34842.51951415037
34741.32803796785
34848.39330677174
34366.98299123207
34857.52157533292
34857.52157533292

2024-04-01 19:54:13.108 Running optimization iteration 1/3, antisymmetrical aberrations34857.86153173289
34856.87304966727
34857.31507785718
34857.40521185808
34857.53192030475
34857.509994402
34856.11695909704
34857.86379585522
34856.99186721925
34856.94402045053
34857.521610966185
34857.520882598285
34857.52157533292
33814.40578349178
33789.408180259656
33839.11679500668
33815.60713305544
33812.88963777555
33814.348363051
33814.46235474518
33860.141488222594
33767.681287372165
33811.28805068883
33816.442402411965
33814.42509945827
33814.38550269116
34857.89003601461
34857.759597554556
34857.71209975514
34857.709740965416
34857.74821559084
34857.899526494046
34857.87982838612
34857.35988349095
34857.358403708415
34857.31335698061
34857.360513578365
34857.89045112888
34857.8891358349
34857.8913431976
34857.7554265043
34857.71927904776
34857.70495510768
34857.75537927224
34857.900556626824
34857.880848750756
34857.37159375773
34857.3491474175
34857.32424752581
34857.350803358284
34857.89117042291
34857.89027276151
34857.89928253053
34857.73398748031
34857.75635207126
34857.683916763184
34857.79253259886
34857.90859820545
34857.88925945282
34857.43388874521
34857.30294267259
34857.387064122806
34857.30480124974
34857.89965266368
34857.89812891974
34857.913277239866
34857.72010540763
34857.798662614536
34857.671203363665
34857.833198867425
34857.922397686925
34857.90331184088
34857.49972490952
34857.265026630776
34857.45117292503
34857.26836096697
34857.91316789324
34857.9124094371
34857.96346414487
34857.707967536626
34857.91050091663
34857.664442899855
34857.93933864119
34857.971167046715
34857.95367745792
34857.66344773147
34857.20091233968
34857.607036222
34857.21273719486
34857.9630684576
34857.962015368925
34858.07736047774
34857.73543896617
34858.11064476199
34857.69953323073
34858.13252334045
34858.0841560142
34858.06963598127
34857.935486187656
34857.15696230184
34857.8709739991
34857.17683178294
34858.07755021089
34858.07607726828
34858.32583034617
34857.87379427396
34858.46947987189
34857.8568707237
34858.47217717816
34858.32885498434
34858.32195813491
34858.382251738454
34857.20688841581
34858.29763381638
34857.247168358466
34858.32558231144
34858.324926734924
34858.73874417041
34858.73874417041

2024-04-01 19:54:14.669 Running optimization iteration 1/3, stage orientation34858.73874417041
34772.0267772083
34887.044662453365
34915.19455471411
34914.92790685259
34918.66858901254
34918.66858901254

2024-04-01 19:54:15.206 Running optimization iteration 1/3, particle poses34918.66858901254
35083.710592262556
35181.746564183675
35222.48135746661
35252.66101055608
35272.23209589749
35287.95883027918
35315.681257414355
35325.60681553673
35347.22957561828
35347.22957561828

2024-04-01 19:54:17.295 Running optimization iteration 2/3, image & volume warping35347.22957561828
35372.8585798554
35352.23410804347
35373.02755323833
35363.55327089579
35372.67768593039
35372.99315335885
35372.99819774683
35373.027519987736
35373.03048931123
35373.03048931123

2024-04-01 19:54:18.314 Running optimization iteration 2/3, antisymmetrical aberrations35372.66495049962
35373.07574532731
35372.83084653005
35372.89596316668
35373.02127111032
35373.03787072469
35372.900593553786
35372.06094686658
35372.49108366023
35372.427027066406
35373.01591816733
35373.043942670774
35373.03048931123
34277.51349572015
34303.20494507726
34251.52630353775
34279.5237998585
34275.17764700537
34277.57206714992
34277.45337899951
34229.084146338246
34324.92514419099
34272.90287306359
34281.010200181925
34277.502276097206
34277.52289368405
35373.10858070242
35372.96339672988
35372.9341590402
35372.92792154891
35372.95516647419
35373.100305978034
35373.11595193487
35372.570936876735
35372.546980789804
35372.534561399734
35372.54006643269
35373.093757610324
35373.12244856484
35373.108719380754
35372.966516590604
35372.932318558866
35372.93043806373
35372.95418594051
35373.10063823977
35373.11643377222
35372.56642970957
35372.552357724686
35372.53144726519
35372.54443263762
35373.094237329344
35373.12294929773
35373.1109419395
35372.978006283054
35372.92433546628
35372.94068025747
35372.94788885142
35373.103146716974
35373.11830903871
35372.549732442654
35372.57264606223
35372.51809553366
35372.561733471994
35373.09638651429
35373.12485737697
35373.12321560825
35373.021460335
35372.906062854345
35372.97750575522
35372.93614493531
35373.11599503654
35373.130451582554
35372.50542016486
35372.64250275843
35372.48615781164
35372.61846687648
35373.10936567837
35373.13697111878
35373.16183699395
35373.11932383851
35372.884432234794
35373.06051510044
35372.92977365992
35373.15436554855
35373.16840123129
35372.432076016376
35372.79271924847
35372.44275083696
35372.73862813037
35373.148754255984
35373.17409435008
35373.243777561845
35373.27824522151
35372.88989156364
35373.19924024246
35372.95401164395
35373.23722850704
35373.24874629465
35372.36816874893
35373.01986115766
35372.420593235816
35372.92415402974
35373.23314495713
35373.25304718512
35373.39612072313
35373.497941502515
35372.97536761501
35373.395933690714
35373.06352099472
35373.3934532967
35373.39862030197
35372.390441738295
35373.302771865056
35372.49723760699
35373.15308623977
35373.391925861324
35373.399685055396
35373.60324169484
35373.60324169484

2024-04-01 19:54:20.894 Running optimization iteration 2/3, stage orientation35373.60324169484
35260.73699101371
35385.15345910717
35400.83605113433
35392.08023110412
35401.2009048056
35401.2009048056

2024-04-01 19:54:21.698 Running optimization iteration 2/3, particle poses35401.2009048056
35407.03062742754
35426.38913876819
35431.71097921788
35441.64169459663
35452.56961916675
35469.08410278201
35485.10876135355
35500.281239708354
35512.40980335207
35512.40980335207

2024-04-01 19:54:24.912 Running optimization iteration 3/3, image & volume warping35512.40980335207
35516.61026569651
35446.23909153196
35514.69355967961
35516.38535061823
35516.55982338489
35516.60443412937
35516.61561559512
35516.61561559512
35516.61026569651
35516.61026569651

2024-04-01 19:54:26.093 Running optimization iteration 3/3, antisymmetrical aberrations35516.48792790982
35516.41134957802
35516.36730619916
35516.5173911012
35516.59209220931
35516.6271502605
35516.01091931638
35516.10348783498
35516.15491469837
35515.91534687534
35516.60960070308
35516.61048802368
35516.61026569651
34370.58312340165
34360.05218482448
34380.80357429216
34395.61231453568
34345.24332342248
34370.523124014835
34370.6423343757
34389.32157281855
34350.7740256329
34323.391821608304
34416.70272488634
34370.610139386874
34370.55468445219
35516.618480790436
35516.4680224291
35516.44702936785
35516.443616516415
35516.45735708867
35516.600523947105
35516.63583636984
35516.07090803176
35516.0597990911
35516.03758916552
35516.04892830102
35516.61734682398
35516.61852872644
35516.61880394234
35516.467840526864
35516.44845763135
35516.44458637935
35516.457409996416
35516.60077484968
35516.63595209486
35516.07264942593
35516.058747458796
35516.03653600239
35516.051447138096
35516.61763569265
35516.61913092923
35516.61973094233
35516.46664501539
35516.45238239607
35516.44862489199
35516.45577139945
35516.60193599834
35516.63759455558
35516.078702759216
35516.05588067278
35516.033245647304
35516.05746694169
35516.61914221868
35516.62054621083
35516.62535420189
35516.46105374354
35516.46819603203
35516.464951641
35516.45006928852
35516.60748245006
35516.64251459477
35516.10352675161
35516.041967483114
35516.01722290157
35516.08362782764
35516.62422121512
35516.62597119813
35516.683451736375
35516.45298038649
35516.59204535978
35516.59432412236
35516.43662234867
35516.66763490699
35516.69787698675
35516.283280754345
35515.97753978835
35515.94504277855
35516.271832353064
35516.68251232046
35516.68325869733
35516.773458531
35516.49605101002
35516.72963696253
35516.747791509384
35516.463812836664
35516.7635082404
35516.78296639308
35516.45771191828
35515.98390373909
35515.92056630367
35516.477217877466
35516.773229146056
35516.77349297564
35516.89508756138
35516.62444351722
35516.84402567887
35516.87860045862
35516.57531435227
35516.892992247514
35516.895992135534
35516.561377481914
35516.12232189199
35516.026815777615
35516.613218470666
35516.89466377139
35516.89412530042
35511.97213669224
35516.89596518209
35516.89596518209

2024-04-01 19:54:29.393 Running optimization iteration 3/3, stage orientation35516.89596518209
35333.38767934713
35517.58033430055
35517.7594692217
35512.62937991429
35518.452154898
35518.452154898

2024-04-01 19:54:30.412 Running optimization iteration 3/3, particle poses35518.452154898
35499.8219963348
35525.11305051848
35528.33243737114
35534.96455095588
35538.516182752974
35542.57769984341
35546.835587526366
35550.44159975453
35558.532913743154
35558.532913743154

2024-04-01 19:54:33.709 Final score: 35558.532913743154

2024-04-01 19:54:33.737 57635 MB after optimization of TS_17.tomostar
2024-04-01 19:54:33.765 Calculating FRC between projections and particles for weight optimization
2024-04-01 19:54:40.279 57767 MB after optimization teardown of TS_17.tomostar
2024-04-01 19:54:40.281 Extracting and back-projecting particles...
2024-04-01 19:54:46.716 58405 MB after backprojection of TS_17.tomostar
2024-04-01 19:54:46.942 62095 MB after full refinement of TS_17.tomostar
2024-04-01 19:54:46.982 Finished refining TS_17.tomostar
2024-04-01 19:54:46.982 Execution took 63.225 seconds
2024-04-01 19:54:46.983 
```

**Comment by alisterburt** [comment]
can't debug properly in our HPC without ending up crashing nodes with out of memory due to restrictions on interactive nodes - will have to move to the windows box

**Comment by alisterburt** [comment]
appears fixed on `2.0.0dev10`
